swig/include/csharp/typemaps_csharp.i

```diff
@@ -598,3 +606,44 @@ OPTIONAL_POD(int, int);
  * GDALGetLayerByName typemaps                                                *
  *****************************************************************************/
 %apply ( const char *utf8_path ) { const char* layer_name };
+
+/******************************************************************************
+ * SpatialReference.FindMatches                                               *
+ *****************************************************************************/
+%apply (int *hasval) {int *nvalues};
+%typemap(imtype, out="IntPtr") OSRSpatialReferenceShadow** FindMatches "SpatialReference[]"
+%typemap(cstype) OSRSpatialReferenceShadow** FindMatches %{SpatialReference[]%}
+%typemap(imtype) int** confidence_values "out IntPtr"
+%typemap(cstype) int** confidence_values %{out int[]%}
+%typemap(csin) int** confidence_values "out confValPtr"
+%typemap(in) (int** confidence_values)
+{
+  /* %typemap(in) (int** confidence_values) */
+  $1 = ($1_ltype)$input;
+}
+%typemap(csout, excode=SWIGEXCODE) OSRSpatialReferenceShadow** FindMatches {
+        /* %typemap(csout) char** FindMatches */
+        IntPtr confValPtr;
+        IntPtr cPtr = $imcall;
+        IntPtr objPtr;
+        SpatialReference[] ret = new SpatialReference[nvalues];
+        confidence_values = (confValPtr == IntPtr.Zero) ? null : new int[nvalues];
+        if (nvalues > 0) {
+	        for(int cx = 0; cx < nvalues; cx++) {
+                objPtr = System.Runtime.InteropServices.Marshal.ReadIntPtr(cPtr, cx * System.Runtime.InteropServices.Marshal.SizeOf(typeof(IntPtr)));
+                ret[cx]= (objPtr == IntPtr.Zero) ? null : new SpatialReference(objPtr, true, null);
```

Comment:
Probably a code comment explaining why OSRFreeSRSArray() is not used could be welcome. I suppose new SpatialReference(objPtr, true) takes ownership of the SRS returned by OSRFindMatches() ?